worker/jobTypes/githubJob.js

```diff
@@ -116,7 +116,8 @@ class GitHubJobClass {
       const exec = workerUtils.getExecPromise();
       const pullRepoCommands = [
         `cd repos/${this.getRepoDirName(currentJob)}`,
-        `git checkout ${currentJob.payload.branchName}`,
+        `git checkout ${currentJob.payload.newHead}`,
+        `git checkout -b ${currentJob.payload.branchName}`,
```

Comment:
Is newHead always populated? What about through the slack app -- is there a condition in which we only have the branch name and latest commit is implied?